content/en/docs/concepts/storage/persistent-volumes.md

```diff
@@ -847,9 +847,9 @@ is turned on.
   PVs of that default. Specifying a default StorageClass is done by setting the
   annotation `storageclass.kubernetes.io/is-default-class` equal to `true` in
   a StorageClass object. If the administrator does not specify a default, the
-  cluster responds to PVC creation as if the admission plugin were turned off. If
-  more than one default is specified, the admission plugin forbids the creation of
-  all PVCs.
+  cluster responds to PVC creation as if the admission plugin were turned off. If more than one
+  default StorageClass is specified, the newest default is used when the
+  PVC is dynamically provisioned.
```

Comment:
Here's what I think Kubernetes is **supposed** to do: when there are two default StorageClasses, neither is treated as the default. As soon as there is only one default StorageClass, that's the one that gets applied.

https://github.com/kubernetes/website/pull/42551 is how I wrote this up - based on my understanding of the _intended_ behavior.

Maybe there is a bug in Kubernetes?

Reply:
Ah, it changed following https://github.com/kubernetes/kubernetes/issues/110514